src/signer.rs

```diff
@@ -254,8 +258,22 @@ impl LightsparkSigner {
         mul_tweak: Option<Vec<u8>>,
     ) -> Result<SecretKey, LightsparkSignerError> {
         let derived_key = self.derive_key(derivation_path).unwrap();
-        let add_tweak: Option<[u8; 32]> = add_tweak.map(|tweak| tweak.try_into().unwrap());
-        let mul_tweak: Option<[u8; 32]> = mul_tweak.map(|tweak| tweak.try_into().unwrap());
+        let add_tweak: Option<[u8; 32]> = add_tweak
```

Comment:
But this will still throw error. We probably want to treat the empty array as nil

Reply:
Chatted offline, but for posterity -- the `filter` call right below will transform empty array to `None`.